DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UISegmentRecorder.swift

```diff
@@ -7,6 +7,14 @@
 import UIKit
 
 internal struct UISegmentRecorder: NodeRecorder {
+    var textObfuscator: (ViewTreeRecordingContext) -> TextObfuscating = { context in
+        switch context.recorder.privacy {
+        case .allowAll:         return context.textObfuscators.nop
+        case .maskAll:          return context.textObfuscators.spacePreservingMask
+        case .maskUserInput:    return context.textObfuscators.spacePreservingMask
+        }
+    }
```

Comment:
The more I look at this the more it feels it should be something like:

```suggestion
    var textObfuscator: (SessionReplayPrivacy) -> TextObfuscating = { privacy in
        switch privacy {
        case .allowAll:         return TextObfuscators.nop
        case .maskAll:          return TextObfuscators.spacePreservingMask
        case .maskUserInput:    return TextObfuscators.spacePreservingMask
        }
    }
```

IMO slightly simpler because we remove dependency between context and obfuscator and highlight the fact that it's chosen based on the privacy property (from context).

Reply:
Accepting `SessionReplayPrivacy` seems simpler at glance, but it enforces **global** and **static** obfuscators that would have to be freed from `SR.deinit()` if they use any state in the future (e.g. for mask caching). We tend to avoid singletons if possible (they're painful in tests).

The `ViewTreeRecordingContext` in turn will come handy for adding fine-grained privacy control later (e.g. different masking per element instance). It could be obvious / simplest choice to pass masking configuration through `context`.

How does it sound @maciejburda ?